ambry-router/src/main/java/com.github.ambry.router/SimpleOperationTracker.java

```diff
@@ -75,11 +75,15 @@
    * @param partitionId The partition on which the operation is performed.
    * @param crossColoEnabled {@code true} if requests can be sent to remote replicas, {@code false}
    *                                otherwise.
+   * @param originatingDcName The original DC where blob was put.
+   * @param includeNonOriginatingDcReplicas if take the option to include remote non originating DC replicas.
+   * @param replicasRequired The number of replicas required for the operation.
```

Comment:
`minReplicasRequired` makes more sense I think. Also, what happens if total number of replicas is less than this number?

Reply:
The logic is to use replicas <= this number, `min*` probably not suitable here. 
It's somehow a `max`. But `max*` is a little confused too as we want exactly this number. 

If total number < this number, it just takes whatever we have. 

```
    if (includeNonOriginatingDcReplicas) {
      replicaPool.addAll(backupReplicas);
      replicaPool.addAll(downReplicas);
    } else {
      // This is for get request only. Take replicasRequired copy of replicas to do the request
      while (replicaPool.size() < replicasRequired && backupReplicas.size() > 0) {
        replicaPool.add(backupReplicas.pollFirst());
      }
      while (replicaPool.size() < replicasRequired && downReplicas.size() > 0) {
        replicaPool.add(downReplicas.pollFirst());
      }
    }
```